stac-example/src/main/scala/geotrellis/stac/raster/StacAssetRasterSource.scala

```diff
@@ -1,28 +1,55 @@
+/*
+ * Copyright 2021 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package geotrellis.stac.raster
 
-import com.azavea.stac4s.StacItemAsset
 import geotrellis.proj4.CRS
 import geotrellis.raster.io.geotiff.OverviewStrategy
 import geotrellis.raster._
 import geotrellis.vector.Extent
-import _root_.io.circe.JsonObject
 import cats.syntax.option._
 
-case class StacAssetRasterSource(
-  asset: StacItemAsset,
-  itemProperties: JsonObject = JsonObject.empty,
-  private[raster] val underlyingRS: Option[RasterSource] = None,
-  private[geotrellis] val targetCellType: Option[TargetCellType] = None
+class StacAssetRasterSource(
+  val asset: StacAsset,
+  private[geotrellis] val targetCellType: Option[TargetCellType],
+  @transient underlyingRS: => Option[RasterSource]
 ) extends RasterSource {
-  private lazy val underlying = underlyingRS.getOrElse(RasterSource(asset.href))
+  @transient private lazy val underlying = underlyingRS.getOrElse(RasterSource(asset.href))
```

Comment:
What is `@transient` doing here? Are we serializing raster sources somewhere?

Reply:
@jisantuc I hope to move it in some form into GT some day, yes.